packages/job-manager/src/batch/batch.ts

```diff
@@ -26,133 +25,139 @@ export class Batch {
 
-        return Array.from(this.jobTasks.values());
+        return tasks;
     }
 
-    public async waitJob(jobId: string, pullIntervalMilliseconds: number = 10000): Promise<void> {
-        this.logger.logInfo(`Waiting for job ${jobId} to complete.`);
+    private async getMaxTasksPerPool(): Promise<number> {
         const client = await this.batchClientProvider();
+        const poolInfo = await client.pool.get(this.config.poolId);
 
-        return new Promise(async (resolve, reject) => {
-            const taskListOptions = {
-                filter: `state ne '${JobTaskState.completed}'`,
-            };
-            const timerId = setInterval(async () => {
-                await client.task
-                    .list(jobId, { taskListOptions: taskListOptions })
-                    .then(async (result: BatchServiceModels.CloudTaskListResult) => {
-                        if (result.length === 0 || (result.length === 1 && result[0].id === process.env.AZ_BATCH_TASK_ID)) {
-                            clearInterval(timerId);
-                            this.logger.logInfo(`Job ${jobId} completed.`);
-                            resolve();
-                        } else {
-                            this.logger.logInfo(`Job ${jobId} in progress with ${result.length} pending tasks.`);
-                        }
-                    })
-                    .catch((error: Error) => {
-                        clearInterval(timerId);
-                        reject(new VError(error, `An error occurred while retrieving the task list for the job ${jobId}`));
-                    });
-                // tslint:disable-next-line: align
-            }, pullIntervalMilliseconds);
-        });
+        return poolInfo.maxTasksPerNode * (poolInfo.currentDedicatedNodes + poolInfo.currentLowPriorityNodes);
     }
 
-    public async getCreatedTasksState(jobId: string): Promise<JobTask[]> {
+    private async getCurrentPoolLoad(): Promise<PoolLoad> {
+        const filterClause = `state eq 'active' and executionInfo/poolId eq '${this.config.poolId}'`;
+        const options = {
+            jobListOptions: { filter: filterClause },
+        };
+
+        const jobs = [];
         const client = await this.batchClientProvider();
-        const cloudTaskListResult = await client.task.list(jobId);
-        this.setTasksState(cloudTaskListResult);
-        let nextLink = cloudTaskListResult.odatanextLink;
-        while (!_.isNil(nextLink)) {
-            nextLink = await this.getTasksStateNext(nextLink);
+        const jobListResponse = await client.job.list(options);
+        jobs.push(...jobListResponse.values());
+
+        let odatanextLink = jobListResponse.odatanextLink;
+        while (odatanextLink !== undefined) {
+            const jobListResponseNext = await client.job.listNext(odatanextLink, options);
+            jobs.push(...jobListResponseNext.values());
+            odatanextLink = jobListResponseNext.odatanextLink;
         }
```

Comment:
move this to a helper function.
getActiveJobs.
P2:
This would also make the testing easier. (to test getCurrentPoolLoad. you can mock getActiveJobs

Reply:
refactored